cterasdk/core/taskmgr.py

```diff
@@ -0,0 +1,83 @@
+import logging
```

Comment:
Is this implementation really different than the one in the Filer? They seem very much alike. If there are some minor differences, we can have a base class and then two sub classes

Reply:
@ygalblum it's possible. I considered that. The amount of shared code will not be big. The Portal task manager is much more complex than the Filer... but I agree that at least the increment/while-loop can be exported to a base class.

Can we proceed with this PR and I will handle this in a separate commit? I don't want to potentially introduce instability and also considering this change may also break some UT. (Opened SDK-59)

Reply:
I just did a diff between the files. There are mainly two differences:
1. The TID method (which can be implemented in the sub class
2. The Portal implementation also has the option to complete with Warning. There is no harm in checking for it in the generic implementation since in the Filer case, it will just not happen.
I don't like leaving turds to be fixed later since most likely they will not be fixed

Reply:
Ok. In that case, i'll use this opportunity to implement a more complete Portal task manager with the ability to stop/resume tasks on servers

Reply:
@ygalblum - I did the simple thing and just extracted the functionality to a base abstract class and inherited from both edge/core. Let me know if we're getting close